src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxKindsService.cs

```diff
@@ -18,4 +25,15 @@ internal interface ISyntaxKindsService : ILanguageService
         /// </summary>
         int LogicalOrExpression { get; }
     }
+
+    internal abstract class AbstractSyntaxKindsService : ISyntaxKindsService
```

Comment:
❔ Why is this needed? Seems like the interface would be sufficient.

Reply:
So this is due to pain in authoring stuff in VB.  In VB you have to have the explicit interface impl form.  This approach allows hte VB impl to just be simple with overrides.  I agree that's marginal.  So i can remove.  But i've found htis much more pleasant from an authoring perspective as things like implement-interface "work", but don't necessarily respect initial order.  So yo have to go manually use it, then move things around.